llvm/lib/Transforms/IPO/FunctionSpecialization.cpp

```diff
@@ -964,7 +970,7 @@ bool FunctionSpecializer::findSpecializations(Function *F, unsigned FuncSize,
         if (LatencySavings < MinLatencySavings * FuncSize / 100)
           return false;
         // Maximum codesize growth.
-        if (FunctionGrowth[F] / FuncSize > MaxCodeSizeGrowth)
+        if ((FunctionGrowth[F] + CodeSizeCost) / FuncSize > MaxCodeSizeGrowth)
           return false;
 
```

Comment:
Here we successfully got past the FuncGrowth check. We know for sure the specialization will be created. We can safely do `FunctionGrowth[F] += SpecSize;` at this point, before returning `true`.